massa-client/src/tests/scenarios.rs

```diff
@@ -1,40 +1,51 @@
+use anyhow::{bail, Result};
 use assert_cmd::Command;
 use serial_test::serial;
-use std::thread;
 
-#[tokio::test]
-#[serial]
-async fn test_if_exit_gracefully() {
-    let mut cmd = Command::cargo_bin("massa-client").unwrap();
-    cmd.arg("exit").assert().success();
+const TIMEOUT: u64 = 5;
+const TENTATIVES: u64 = 5;
+
+async fn run_client_cmd(cmd: &str) -> Result<String> {
+    for _ in 0..TENTATIVES {
+        let output = std::str::from_utf8(
+            &Command::new("cargo")
+                .args(&["run", "--", cmd, "--json"])
+                .assert()
+                .get_output()
+                .stdout,
+        )?
+        .to_string();
+        // TODO: compare with tcp connect error as a JSON object?
+        if !output.contains("tcp connect error") {
+            return Ok(output);
+        }
+        tokio::time::sleep(std::time::Duration::from_secs(TIMEOUT)).await;
+    }
+    bail!("tcp connect error")
 }
 
-const CONFIG_PATH: &str = "../massa-node/src/tests/config.toml";
-const TIMEOUT: u64 = 5;
 #[tokio::test]
 #[serial]
-async fn test_if_node_stop() {
-    let massa_node_thread_handle = thread::spawn(|| {
-        Command::cargo_bin("massa-node")
+async fn run_node() {
+    // TODO: @adrien-zinger do we have a better way to spawn `massa_node`?
+    std::thread::spawn(|| {
+        Command::new("cargo")
+            .arg("run")
+            .current_dir("../massa-node")
             .unwrap()
```

Comment:
I'm afraid that this could run since the end of time. How do you replace the timeout?

Reply:
What good value did you advise for a timeout?

Reply:
If you go for the full scenario, I would tell 10 minutes, given that genesis timestamp is between 3 and 5 minutes in the future

Reply:
I you just test if the node start and run, a timeout will run in error whatever you do. I think that the test for stoping the node is better.

Reply:
the full scenario here #1347 finish by stopping the node, so that should not be an issue :)

Reply:
what if you failed stoping the node?